src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java

```diff
@@ -303,20 +304,28 @@ public void afterNavigateTo(String url, WebDriver driver) {
     Class<?> declaringClass = method.getDeclaringClass();
 
     if (TestContext.isFirstLoad()) {
-      User user = null;
       TestContext.setFirstLoad(false);
+    }
 
-      if (declaringClass.isAnnotationPresent(Execute.class)) {
-        user = declaringClass.getAnnotation(Execute.class).asUser();
-      }
+    // SUS-1737 / QAART-937: Perform this check on every page load
+    // to check if user is not logged out
+    User user = User.ANONYMOUS;
 
-      if (method.isAnnotationPresent(Execute.class)) {
-        user = method.getAnnotation(Execute.class).asUser();
-      }
+    if (declaringClass.isAnnotationPresent(Execute.class)) {
+      user = declaringClass.getAnnotation(Execute.class).asUser();
+    }
 
-      if (user != null && user != User.ANONYMOUS) {
+    if (method.isAnnotationPresent(Execute.class)) {
+      user = method.getAnnotation(Execute.class).asUser();
+    }
+
+    if (user != User.ANONYMOUS) {
+      WikiBasePageObject basePageObject = new WikiBasePageObject();
+      try {
+        basePageObject.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)